plugins/available/alias-completion.plugin.bash

```diff
@@ -52,7 +52,7 @@ function alias_completion {
 		read -a alias_arg_words <<< "$alias_args"
 
 		# skip alias if there is no completion function triggered by the aliased command
-		if [[ ! " ${completions[*]} " =~ $alias_cmd ]]; then
+		if [[ ! " ${completions[*]} " == " $alias_cmd " ]]; then
```

Comment:
Now that I've had a minute to look here, what I *think* we're trying to do is :
* Determine if the array variable `$completions` contains any element with the exact value contained in the scalar variable `$alias_cmd`

Does that sound right?  Assuming yes ...

Testing this syntax in isolation seemed to fail for me, and I can't find any bash documentation that suggests `[[  "${ARRAY[*]}" == "VALUE" ]]` is a valid technique to check if an array contains a specific value.

Although the answers in this thread are very close:
* [Check if an element is present in a Bash array](https://stackoverflow.com/questions/14366390/check-if-an-element-is-present-in-a-bash-array)

This thread mentions looping, which is likely the safest technique, but does also include an a clever regex+IFS solution (lower in the answer list):
* [Check if a Bash array contains a value](https://stackoverflow.com/questions/3685970/check-if-a-bash-array-contains-a-value)

I did also find this EXTGLOB based technique is is somewhat similar:
* [Checking if an array contains an element in bash](https://dev.to/meleu/checking-if-an-array-contains-an-element-in-bash-5bn1) (compatible with Bash 2.02+

So there's a few techniques to choose from - and maybe the technique provided works but I tested it wrong, but it sSeems like a simple function that takes a value and an array and loops over the array looking for the value would be most accurate.

Reply:
hmm, very interesting.

do you think we should change this implementation then?
I want to get it fixed, as it is quite an annoying bug

Reply:
Yes, I think the safest solution would probably be to introduce a library function, say `_array_contains` and have it loop over the array, checking each value for a match.

Reply:
I looked around and saw we already have this kind of utility (_bash-it-array-contains-element)

See now :smile: